draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -831,16 +834,82 @@ Instantiating one of these protocols with an elliptic curve group motivates
-hash2curve(alpha) is not sufficient to get uniformity, however it can be used
-as a building block for obtaining a random oracle as is described in {{rom}}.
+h(x) is not sufficient to get uniformity and cannot be used as a random oracle.
+
+Brier et al. {{BCIMRT10}} describe two generic constructions that give a
+hash function approximating a random oracle. Farashahi et al. {{FFSTV13}}
+and Tibouchi and Kim {{TK17}} refine this analysis. In particular, Farashahi
+et al. show that summing two independent evaluations of any of the
+deterministic endings of {{encodings}} suffices to approximate a random
+oracle to an elliptic curve. In other words:
+
+~~~
+   h(x) = f(H0(x)) + f(H1(x))
+~~~
+
+where f: F -> E(F) is a deterministic encoding, H0 and H1 are hash
+functions (modeled as random oracles) mapping strings to elements of F, and the
+addition operation corresponds to point addition on the target elliptic curve.
+
+
+# Roadmap on Hash to Curve. {#roadmap}
+
+The following functions provide a general framework for mapping bit
+strings into points of a prime group G.
+
+ - Injective encodings. A bit string is mapped to a point in G, however,
+   the distribution of this function is not uniform.
+
+   ~~~
+   h2cInjective(alpha)
+
+   Input: alpha, an arbitrary-length bit string.
+   Output: P, a point in the group G.
+
+   Steps:
+   1. u = hash2base(alpha, I2OSP(0, 1)) // {0,1}* -> F   
+   2. Q = map2curve(u)                  //   F    -> E(F)
+   3. P = clearCofactor(Q)              //  E(F)  -> G   
+   4. Output P
+   ~~~
+
+ - Random Oracle. Unlike injective encodings, this function can be used as a
+   random oracle of points.
+
+   ~~~
+   h2cRandomOracle(alpha)
+
+   Input: alpha, an arbitrary-length bit string.
+   Output: P, a point in the group G.
+
+   Steps:
+   1. u0 = hash2base(alpha, I2OSP(0, 1))  //    {0,1}*   -> F   
+   2. u1 = hash2base(alpha, I2OSP(1, 1))  //    {0,1}*   -> F   
+   3. Q0 = map2curve(u0)                  //      F      -> E(F)
+   4. Q1 = map2curve(u1)                  //      F      -> E(F)
+   5. R = Q0 + Q1                         // E(F) x E(F) -> E(F)
+   6. P = clearCofactor(R)                //     E(F)    -> G   
+   7. Output P
+   ~~~
+
+The pieces used for constructing these functions hold particular properties,
+which are described in the subsequent sections. First, {{utility}} introduces
+some utility functions that are used across the document. {{hashtobase}}
+describes the hash2base function. Due to the diversity of elliptic curve models
+there exist several ways construct points from a field elements. Some
+possible choices are described in {{encodings}}. Similarly, clearing the
+cofactor is particular to the group G, and this is discussed in {{cofactor-clearing}}.
+
+{{suites}} of this document provides a list of suites for hashing to curve.
+Each suite can be considered as a full set of parameters and algorithms required
```

Comment:
It occurs to me that there are no suites for injective encodings. Should we add some?

Reply:
Stashed a note about injective encodings suites in #126.